interop/ksp/test-processor/src/test/kotlin/com/squareup/kotlinpoet/ksp/test/processor/TestProcessorTest.kt

```diff
@@ -300,8 +300,8 @@ class TestProcessorTest {
       public class Example {
         public fun aliases(
           aliasedName: String,
-          genericAlias: List<String>,
-          genericMapAlias: Map<Int, String>,
+          genericAlias: List,
```

Comment:
@ZacSweers this looks like a regression in KSP?

Reply:
yes we need to fix something on the kotlinpoet side. Behavior change rather than a regression

Reply:
I can take a look.

Reply:
https://github.com/square/moshi/issues/1530

Reply:
@ZacSweers PTAL! https://github.com/square/kotlinpoet/pull/1240/commits/abbcc178857ef4956d0c6cff883a9790cd6a9c58